testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/workmanager/distributed/DwmWatermarkTestCase.java

```diff
@@ -0,0 +1,152 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.manualmode.jca.workmanager.distributed;
+
+import javax.naming.NamingException;
+import javax.resource.spi.work.Work;
+import javax.resource.spi.work.WorkException;
+import java.io.IOException;
+
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.logging.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests distributed work manager and whether it really distributes work over multiple nodes. Test cases use two servers
+ * both with a deployed resource adapter configured to use the DWM. Tests with WATERMARK policy.
+ */
+@ServerSetup(DwmWatermarkTestCase.DwmWatermarkServerSetupTask.class)
+@RunWith(Arquillian.class)
+@RunAsClient
+public class DwmWatermarkTestCase extends AbstractDwmTestCase {
+
+    private static Logger log = Logger.getLogger(DwmWatermarkTestCase.class.getCanonicalName());
+
+    private static final int WATERMARK_MAX_THREADS = 2;
+
+    static class DwmWatermarkServerSetupTask extends AbstractDwmTestCase.DwmServerSetupTask {
+        @Override
+        protected Policy getPolicy() {
+            return Policy.WATERMARK;
+        }
+
+        @Override
+        protected Selector getSelector() {
+            return Selector.MAX_FREE_THREADS;
+        }
+
+        @Override
+        protected int getWatermarkPolicyOption() {
+            return WATERMARK_MAX_THREADS - 1;
+        }
+
+        @Override
+        protected int getSrtMaxThreads() {
+            return WATERMARK_MAX_THREADS;
+        }
+
+        @Override
+        protected int getSrtQueueLength() {
+            return WATERMARK_MAX_THREADS;
+        }
+    }
+
+    /**
+     * Executes a long work instances on a single node and verifies that it took enough time and executed on the
+     * expected node (this one since Policy.WATERMARK)
+     */
+    @Test
+    @InSequence(11)
+    public void testDoWork() throws IOException, NamingException, WorkException, InterruptedException {
+        log.info("Running testDoWork()");
+
+        long startTime = System.currentTimeMillis();
+        int doWorkAccepted = server1Proxy.getDoWorkAccepted();
+        int distributedDoWorkAccepted = server2Proxy.getDistributedDoWorkAccepted();
+
+        server1Proxy.doWork(new LongWork().setName("testDoWork-work1"));
+
+        Assert.assertTrue("Expected time >= " + (startTime + LongWork.WORK_TIMEOUT) + ", actual: " + System.currentTimeMillis(),
+                startTime + LongWork.WORK_TIMEOUT <= System.currentTimeMillis());
+
+        Thread.sleep(LongWork.WORK_TIMEOUT + 100); // wait for the scheduled work to start and finish, so it doesn't mess up our statistics for other tests
```

Comment:
@ctomc I removed all logging as requested, I'll push when the sleep change is ready.

@rsvoboda The InSequence is only really needed for Watermark test. I removed it from the other tests. The reason Watermark test needs it is that it is a workaround for a specific behavior of thread pools that needs to be documented (and is not a bug), I guess @maeste could explain that in detail if need be. There is a different way to work around it, but this is much easier. Running any test method separately works without issues, the two existing methods in Watermark test only influence themselves when run in certain order.

Regarding the sleep, yes, I don't know when exactly the work finishes, that's why it's there. I'm not against removing it, I just don't have a different solution that I'm sure will work.

I tried to keep the sleep to minimum (only while the work is being executed) and I can further cut it by lowering the work duration. I can get it from <60 sec for the three classes to <20 sec.

How should I proceed?